src/experimental/tools/download2go/index.ts

```diff
@@ -0,0 +1,208 @@
+  IOptionsStarter,
+  IProgressBarBuilderAbstract,
+  IPublicAPI,
+  IStoredManifest,
+} from "./types";
+
+export default async function D2G({
+  nameDB = "d2g",
+  storeManifestEvery,
+}: IOptionsStarter = {}): Promise<IPublicAPI> {
+  const emitter = new EventEmitter<IEventsEmitter>();
+  const db = await setUpDb(nameDB);
+  if (!db) {
+    throw new IndexDBError("A problem occured during the set up of IndexDB");
+  }
+  const activeSubsDownloader: IActiveSubs = {};
+  const activePauseSubject: IPauseSubject = {};
+
+  return {
+    emitter,
+    async download(settings: IAddMovie): Promise<any> {
+      try {
+        await checkForSettingsAddMovie(settings, db, activeSubsDownloader);
+        const size = 0;
+        const progressBarBuilder = {
+          progress: 0,
+          overall: 0,
+          downloadedID: [],
+        };
+        const progress$ = new ReplaySubject<IProgressBarBuilderAbstract>(1);
+        const pause$ = new AsyncSubject<void>();
+        const { contentID } = settings.dbSettings;
+        activePauseSubject[contentID] = pause$;
+        const subcriptionDownloader = downloader(
+          { ...settings, type: "start" },
+          {
+            activeSubsDownloader,
+            pause$,
+            progress$,
+            progressBarBuilder,
+            size,
+          },
+          { db, emitter, storeManifestEvery }
+        );
+        activeSubsDownloader[contentID] = subcriptionDownloader;
+      } catch (e) {
+        emitter.trigger("error", {
+          action: "download",
+          contentID: settings.dbSettings.contentID,
+          error: e || new Error("A Unexpected error happened"),
+        });
+        return undefined;
+      }
+    },
+    async resume(contentID: string): Promise<void> {
+      try {
+        const manifest: IStoredManifest = await db.get("manifests", contentID);
+        checkForResumeAPausedMovie(manifest, activeSubsDownloader);
+        const size = manifest.size;
+        const progressBarBuilder = manifest.progressBarBuilder;
+        const progress$ = new ReplaySubject<IProgressBarBuilderAbstract>(1);
+        const pause$ = new AsyncSubject<void>();
+        const settings = {
+          contentID,
+          metaData: manifest.metaData,
+          rxpManifest: manifest.rxpManifest,
+        };
+        activePauseSubject[contentID] = pause$;
+        const subcriptionDownloader = downloader(
+          { ...settings, type: "resume" },
+          {
+            activeSubsDownloader,
+            pause$,
+            progress$,
+            progressBarBuilder,
+            size,
+          },
+          { db, emitter, storeManifestEvery }
+        );
+        activeSubsDownloader[contentID] = subcriptionDownloader;
+      } catch (e) {
+        emitter.trigger("error", {
+          action: "resume",
+          contentID,
+          error: e || new Error("A Unexpected error happened"),
+        });
+      }
+    },
+    pause(contentID: string): number | void {
```

Comment:
Not a huge fan of returning number as a result. It should be more meaningful.

Reply:
Not a fan neither. You could return an enum or descriptive strings/objects instead.

Reply:
yea, I just deleted any return and will return void, I think it is not necessary to notify the user here.